packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,341 @@
+   * command override. You can also override existing environment variables
+   * (that are specified in the job definition or Docker image) on a container
+   * or add new environment variables to it with an environment override.
+   *
+   * @default - No container overrides
+   */
+  readonly containerOverrides?: ContainerOverrides;
+
+  /**
+   * A list of dependencies for the job.
+   * A job can depend upon a maximum of 20 jobs. You can specify a SEQUENTIAL
+   * type dependency without specifying a job ID for array jobs so that each
+   * child array job completes sequentially, starting at index 0.
+   * You can also specify an N_TO_N type dependency with a job ID for array jobs.
+   * In that case, each index child of this job must wait for the corresponding
+   * index child of each dependency to complete before it can begin.
+   *
+   * @default - No dependencies
+   */
+  readonly dependsOn?: JobDependency[];
+
+  /**
+   * The payload to be passed as parametrs to the batch job
+   *
+   * @default - No parameters are passed
+   */
+  readonly payload?: { [key: string]: any };
+
+  /**
+   * The retry strategy to use for failed jobs from this SubmitJob operation.
+   * When a retry strategy is specified here,
+   * it overrides the retry strategy defined in the job definition.
+   *
+   * @default - No retry strategy
+   */
+  readonly retryStrategy?: RetryStrategy;
+
+  /**
+   * The timeout configuration for this SubmitJob operation.
+   *
+   * @default - No timeout
+   */
+  readonly timeout?: Duration;
+
+  /**
+   * The service integration pattern indicates different ways to call TerminateCluster.
+   *
+   * The valid value is either FIRE_AND_FORGET or SYNC.
+   *
+   * @default SYNC
+   */
+  readonly integrationPattern?: sfn.ServiceIntegrationPattern;
+}
+
+/**
+ * A Step Functions Task to run AWS Batch
+ *
+ * @experimental
+ */
+export class RunBatchJob implements sfn.IStepFunctionsTask {
+  private readonly integrationPattern: sfn.ServiceIntegrationPattern;
+
+  constructor(private readonly props: RunBatchJobProps) {
+    this.integrationPattern =
+      props.integrationPattern || sfn.ServiceIntegrationPattern.SYNC;
+
+    const supportedPatterns = [
+      sfn.ServiceIntegrationPattern.FIRE_AND_FORGET,
+      sfn.ServiceIntegrationPattern.SYNC
+    ];
+
+    if (!supportedPatterns.includes(this.integrationPattern)) {
+      throw new Error(
+        `Invalid Service Integration Pattern: ${this.integrationPattern} is not supported to call RunBatchJob.`
+      );
+    }
+  }
+
+  public bind(_task: sfn.Task): sfn.StepFunctionsTaskConfig {
+    return {
+      resourceArn: getResourceArn(
+        'batch',
+        'submitJob',
+        this.integrationPattern
+      ),
+      policyStatements: [
+        new iam.PolicyStatement({
+          resources: ['*'],
+          actions: [
+            'batch:SubmitJob',
+            'batch:DescribeJobs',
+            'batch:TerminateJob'
+          ]
```

Comment:
Can we restrict `resources` here to just the batch jobs ARNs and avoid `*`? Can we drop the `TerminateJob` policy?

One of the rules we follow in the CDK is to provide the least privilege permissions.

Reply:
AWS Batch does not support resource-level access control. [Reference](https://docs.aws.amazon.com/step-functions/latest/dg/batch-iam.html)

Reply:
Please add a comment to this effect and a link to this doc.

Reply:
Further, do we need Describe and Terminate?

Reply:
Docs do not explicitly mention about the required actions but I would guess we would definitely require Describe. We can probably drop Terminate. I will try to do integ test by dropping them.

Reply:
Just confirmed that the deployment succeeded without the Describe and Terminate actions. Will remove them with next revision.